### DISCUSSION TITLE
An Instagram like snippet selector

### DISCUSSION THREAD

**Discussion by jgrayla** [katspaugh/wavesurfer.js#3505] [Q&A]
I'm building an audio snippet selector with a similar UX to the Instagram selector. The snippet area (region) stays fixed size and centered while dragging left/right move the waveform.

I've been able to use a region to create the behavior of playing from the start of the snippet when it changes, looping at the end, etc.

In struggling to figure out how I can make drag events on the waveform scroll the waveform, keep the region fixed, and get this type of behavior that is common on social apps today. Video example below. Much thanks!

https://github.com/katspaugh/wavesurfer.js/assets/213323/b8e836e6-b70f-468b-9f1e-6f95f3c82923

**Comment by katspaugh** [comment]
I would recommend using wavesurfer only to render a waveform in a wide container (set the width option as desired), and implement the overlay and scrolling in your app. Regions aren’t designed for this use case.
